vehicles/controller/carmovementcontroller.h

```diff
@@ -39,7 +40,7 @@ class CarMovementController : public MovementController
     QSharedPointer<MotorController> mMotorController;
     QSharedPointer<ServoController> mServoController;
     double mSpeedToRPMFactor = 4123.3; // default for Traxxas Slash VXL
-
+    QSharedPointer<VehicleLighting> mVehicleLighting;
```

Comment:
From my point of view, changing CarMovementController for adding lighting (or any other feature) should have a very good justification and I do not think that VehicleLighting has one. I think that CarMovementContoller should not be changed. 
Instead, VehicleLighting should implement a timer to control the timing itself. It should further get a SharedPointer to the VehicleState where it periodically can check the state of the vehicle and set the lighting accordingly. This way, VehicleLighting would only depend on VehicleState and work for any vehicle without requiring adaptations of the specific vehicle (car, diffdrive, ...).

Reply:
It now uses a SharedPointer to VehicleState where it, by use of internal timer, periodically checks state of vehicle